browser/brave_shields/cookie_expiry_browsertest.cc

```diff
@@ -0,0 +1,228 @@
+  std::vector<net::CanonicalCookie> all_cookies =
+      GetAllCookiesDirect(browser());
+  EXPECT_EQ(1u, all_cookies.size());
+  for (const net::CanonicalCookie& cookie : all_cookies) {
+    EXPECT_LE(
+        (base::Time::Now() + less_than_max - cookie.ExpiryDate()).magnitude(),
+        kMarginForTesting);
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(CookieExpirationTest,
+                       CheckExpiryForDocumentCookieMoreThanMax) {
+  ASSERT_TRUE(https_server_->Start());
+
+  GURL url = https_server_->GetURL("a.com", "/simple.html");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+  JSDocumentCookieWriteCookie(
+      browser(), "max-age=" + std::to_string(k4YearsInDays.InSeconds()));
+  std::vector<net::CanonicalCookie> all_cookies =
+      GetAllCookiesDirect(browser());
+  EXPECT_EQ(1u, all_cookies.size());
+  for (const net::CanonicalCookie& cookie : all_cookies) {
+    EXPECT_EQ((cookie.ExpiryDate() - cookie.CreationDate()).InDays(), 7);
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(CookieExpirationTest,
+                       CheckExpiryForCookieStoreLessThanMax) {
+  ASSERT_TRUE(https_server_->Start());
+  auto less_than_max = base::Days(2);
+  GURL url = https_server_->GetURL("a.com", "/simple.html");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+  JSCookieStoreWriteCookie(browser(),
+                           std::to_string(less_than_max.InMilliseconds()));
+
+  std::vector<net::CanonicalCookie> all_cookies =
+      GetAllCookiesDirect(browser());
+  EXPECT_EQ(1u, all_cookies.size());
+  for (const net::CanonicalCookie& cookie : all_cookies) {
+    EXPECT_LE(
+        (base::Time::Now() + less_than_max - cookie.ExpiryDate()).magnitude(),
+        kMarginForTesting);
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(CookieExpirationTest,
+                       CheckExpiryForCookieStoreMoreThanMax) {
+  ASSERT_TRUE(https_server_->Start());
+
+  GURL url = https_server_->GetURL("a.com", "/simple.html");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+  JSCookieStoreWriteCookie(browser(),
+                           std::to_string(k4YearsInDays.InMilliseconds()));
+
+  std::vector<net::CanonicalCookie> all_cookies =
+      GetAllCookiesDirect(browser());
+  EXPECT_EQ(1u, all_cookies.size());
+  for (const net::CanonicalCookie& cookie : all_cookies) {
+    EXPECT_EQ((cookie.ExpiryDate() - cookie.CreationDate()).InDays(), 7);
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(CookieExpirationTest,
+                       CheckExpiryForHttpCookiesLessThanMax) {
+  auto less_than_max = base::Days(30);
+  std::string cookie_string =
+      "test=http;max-age=" + std::to_string(less_than_max.InSeconds());
+  https_server_->RegisterRequestHandler(base::BindLambdaForTesting(
+      [&](const net::test_server::HttpRequest& request)
+          -> std::unique_ptr<net::test_server::HttpResponse> {
+        auto response = std::make_unique<net::test_server::BasicHttpResponse>();
+        response->AddCustomHeader("Set-Cookie", cookie_string);
```

Comment:
embedded test server has always enabled `/set-cookie` endpoint to do exactly this.

impl:
https://github.com/chromium/chromium/blob/8def63071324792533c0db180cd35c309a9c623a/net/test/embedded_test_server/default_handlers.cc#L239-L257

example:
https://github.com/chromium/chromium/blob/76f28918e8af7b367f384433f9d55f4e9b76b1b0/chrome/browser/content_settings/content_settings_browsertest.cc#L534-L536